### DISCUSSION TITLE
[Alerts] Connect to Slack

### DISCUSSION THREAD

**Discussion by hypernovas** [langfuse/langfuse#1512] [Support] [answered]
Hi guys, this is a super awesome tool, and our team has started to use it for our daily Ops. One thing I would love to learn more is alerting. Can we connect the dashboard monitoring to Slack? We usually have an oncall channel where all the alerts are raised there.

**Comment by clemra** [comment, marked as answer]
Hi @hypernovas -- thank you for the suggestion. 

We have thread on [Webhooks in Langfuse](https://github.com/orgs/langfuse/discussions/1033) which would allow you to push events to Slack.

**Reply by chrisshaw** [reply]
Having looked at the linked concept, I wonder if this is really the answer. Not sure if this is what @hypernovas had in mind, but personally, what we're looking for is closer to getting alerted along a metric, not an event. Similar to hitting various thresholds in New Relic or Sentry.

As a concrete example, suppose we're using a large language model to look for abusive behavior. We want to know that it's continuing to perform its task adequately as models and behaviors drift. So we set up an evaluation (let's say a model-based eval or a custom eval) that runs on our generations. Every individual trace's score is going to be variable, but we want to know if something is seriously wrong with our LLM app if it falls beneath a certain average score, aggregated over the last hour. That's when we want to get alerted (whether it's slack or general webhook is less important).

**Reply by maxdeichmann** [reply]
@chrisshaw thanks a lot! We launched the slack integration for prompt events recently (https://langfuse.com/changelog/2025-07-30-slack-prompt-notifications).
I know that you want to have score based trigger for these integrations. We have this very high up on our roadmap. We need to complete some more infrastructure work in the background to enable such alerts at scale unfortunately.

**Reply by marksalpeter** [reply]
Hi everyone! Our new [Monitors](https://cloud.langfuse.com/project/~/monitors) feature just shipped in cloud beta. You can set threshold alerts on metrics like cost, scores, and latency and route them to Slack, webhooks, or GitHub Actions ([docs](https://langfuse.com/docs/metrics/features/monitors)).

Monitors can be configured directly in the UI. No CloudWatch workarounds needed.

If you have any questions or feedback, please let us know.

cc: @chrisshaw, @Othmane796, @kinano, @anshmove37, @BrendanRitter

**Comment by Othmane796** [comment]
any updates or thoughts on what @chrisshaw  mentioned ? The need is around alerting not webhooks. Looked through the open issues and roadmap but doesn't seem to be mentioned anywhere

**Reply by kinano** [reply]
> any updates or thoughts on what @chrisshaw mentioned ? The need is around alerting not webhooks. Looked through the open issues and roadmap but doesn't seem to be mentioned anywhere

+1 to this. It will be ideal if we could setup:
- custom monitors and metrics e.g. failed evals in the last 1 hour
- fire alerts to Slack if the % of failed evals exceeds a certain threshold
- The only workaround I can think of right now is to wire up custom CloudWatch metric alerts in our log groups (custom code/custom monitors).

**Comment by anshmove37** [comment]
Yes interested in custom alerting as well. For example, if a certain trace score is beyond/less than an acceptable score. And preferably setting this up on the UI directly @maxdeichmann 
If there is an updated timeline on this feature would love to know it.

**Comment by BrendanRitter** [comment]
Also interested in this. The current state of this requires a parallel system like cloudwatch metrics to handle what should be in Langfuse's purview (Judge score alerting)
